File: lib/java/opentoken-cli/src/main/java/com/truveta/opentoken/cli/util/StringMaskingUtil.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.cli.util;

/**
 * Utility class for masking sensitive strings in CLI commands.
 * <p>
 * This class provides methods to mask secrets and sensitive data
 * for logging and display purposes.
 */
public final class StringMaskingUtil {

    private static final String NULL_MASK = "<null>";
    private static final String DEFAULT_MASK = "***";

    private StringMaskingUtil() {
        // Utility class; prevent instantiation.
    }

    /**
     * Masks a sensitive string for logging.
     * <p>
     * - Null values are masked as "&lt;null&gt;"
     * - Strings with 3 or fewer characters are masked as "***"
     * - Longer strings show first 3 characters followed by asterisks
     *
     * @param input The string to mask
     * @return The masked string
     */
    public static String maskString(String input) {
        if (input == null) {
            return NULL_MASK;
        }
        if (input.length() <= 3) {
            return DEFAULT_MASK;
        }
        return input.substring(0, 3) + "*".repeat(input.length() - 3);
    }
}

File: lib/python/openlinktoken-cli/src/main/openlinktoken_cli/util/string_masking_util.py
# SPDX-License-Identifier: MIT
"""
Utility class for masking sensitive strings in CLI commands.
"""

from __future__ import annotations


class StringMaskingUtil:
    """Utility methods for masking sensitive strings."""

    NULL_MASK = "<null>"
    DEFAULT_MASK = "***"

    @staticmethod
    def mask_string(input_str: str | None) -> str:
        """
        Mask a sensitive string for logging.

        - None values are masked as "<null>"
        - Strings with 3 or fewer characters are masked as "***"
        - Longer strings show first 3 characters followed by asterisks

        Args:
            input_str: The string to mask.

        Returns:
            The masked string.
        """
        if input_str is None:
            return StringMaskingUtil.NULL_MASK
        if len(input_str) <= 3:
            return StringMaskingUtil.DEFAULT_MASK
        return input_str[:3] + "*" * (len(input_str) - 3)


def mask_string(input_str: str | None) -> str:
    """Mask a sensitive string for logging."""
    return StringMaskingUtil.mask_string(input_str)
